client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -388,13 +414,13 @@ private void submitRetryPushMergedData(
       int mapId,
       int attemptId,
       ArrayList<DataBatches.DataBatch> batches,
-      StatusCode cause,
       Integer oldGroupedBatchId,
       ReviveRequest[] reviveRequests,
       int remainReviveTimes,
       long reviveResponseDueTime) {
     HashMap<Pair<String, String>, DataBatches> newDataBatchesMap = new HashMap<>();
     ArrayList<DataBatches.DataBatch> reviveFailedBatchesMap = new ArrayList<>();
+    ArrayList<StatusCode> reviveFailedBatchesCauses = new ArrayList<>();
```

Comment:
Use `List` instead, In Java, we prefer to decouple your code from a specific implementation of the interface